controllers/templatesync/template_sync.go

```diff
@@ -110,6 +113,16 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{}, err
 	}
 
+	// Set boolean to handle finalizer
+	var needsFinalizerCleanup bool
+	var finalizerCleanupFailed bool
+
+	if instance.DeletionTimestamp != nil {
+		reqLogger.Info("Policy marked for deletion--proceeding with finalizer cleanup")
+
+		needsFinalizerCleanup = true
```

Comment:
When the parent policy is being deleted, then I think this reconcile should *only* delete the cluster-scoped policy templates and then remove the finalizer. It's currently doing a lot of work that isn't necessary since the policy templates will just be deleted after the reconcile is over.

Reply:
Good point! I'll adjust it.

Reply:
I've created a cleanup flow and function that gets called before any template reconciles.